superset-frontend/src/components/Menu/MenuObject.tsx

```diff
@@ -17,20 +17,29 @@
  * under the License.
  */
 import React from 'react';
-import PropTypes from 'prop-types';
 import { NavItem, NavDropdown, MenuItem } from 'react-bootstrap';
 
-const propTypes = {
-  label: PropTypes.string.isRequired,
-  icon: PropTypes.string.isRequired,
-  index: PropTypes.number.isRequired,
-  url: PropTypes.string,
-  childs: PropTypes.arrayOf(
-    PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
-  ),
-};
+export interface MenuObjectChildProps {
+  label: string;
+  icon: string;
+  index: number;
+  url?: string;
+}
+export interface MenuObjectProps {
+  label?: string;
+  icon?: string;
+  index: number;
+  url?: string;
+  childs?: (MenuObjectChildProps | string)[];
```

Comment:
I'm not sure if this matches what was in the PropTypes before, can you validate that it shouldn't be:
```typescript
childs?: MenuObjectChildProps[] | string[];
```

Reply:
@etr2460 , it creates an issue when using 'map' , since map of ' MenuObjectChildProps[] | string[] ' is not callable.
What do you suggest?

Reply:
ah I see. I think this is fine as is then